packages/fx-core/src/plugins/resource/frontend/plugin.ts

```diff
@@ -255,23 +260,35 @@ export class FrontendPluginImpl {
 
     if (solutionSettings?.activeResourcePlugins?.includes(DependentPluginInfo.RuntimePluginName)) {
       addToEnvs(
-        EnvironmentVariables.RuntimeEndpoint,
+        EnvKeys.RuntimeEndpoint,
         ctx.envInfo.state
           .get(DependentPluginInfo.RuntimePluginName)
           ?.get(DependentPluginInfo.RuntimeEndpoint) as string
       );
-      addToEnvs(EnvironmentVariables.StartLoginPage, DependentPluginInfo.StartLoginPageURL);
+      addToEnvs(EnvKeys.StartLoginPage, DependentPluginInfo.StartLoginPageURL);
     }
 
     if (solutionSettings?.activeResourcePlugins?.includes(DependentPluginInfo.AADPluginName)) {
       addToEnvs(
-        EnvironmentVariables.ClientID,
+        EnvKeys.ClientID,
         ctx.envInfo.state
           .get(DependentPluginInfo.AADPluginName)
           ?.get(DependentPluginInfo.ClientID) as string
       );
     }
+    return envs;
+  }
+
+  private async updateMultiEnv(ctx: PluginContext): Promise<void> {
```

Comment:
I don't quite fully understand these two method of updateMultiEnv and updateDotenv, will check with you offline next week